hpx/lcos/base_lco_with_value.hpp

```diff
@@ -82,6 +83,17 @@ namespace hpx { namespace lcos
 
         virtual void set_event()
         {
+            set_event_nonvirt(std::is_default_constructible<RemoteResult>());
+        }
+
+        void set_event_nonvirt(std::false_type)
+        {
+            // this shouldn't ever be called
+            HPX_ASSERT(false);
```

Comment:
I think an exception would be more appropriate here, I guess. The ideal solution would be to disable `set_event` for non default constructible types alltogether, I think.